xds/src/main/java/io/grpc/xds/XdsClusterResource.java

```diff
@@ -241,6 +241,12 @@ private static StructOrError<CdsUpdate.Builder> parseNonAggregateCluster(
       if (!edsClusterConfig.getServiceName().isEmpty()) {
         edsServiceName = edsClusterConfig.getServiceName();
       }
+      // edsServiceName is required if the CDS resource has an xdstp name.
+      if ((edsServiceName == null || edsServiceName.isEmpty())
```

Comment:
nit: The `edsServiceName` is either null or non-empty string, so `edsServiceName.isEmpty()` is redundant.

Reply:
Good point.  Removed it.